source/brailleDisplayDrivers/hidBrailleStandard.py

```diff
@@ -226,13 +239,20 @@ def display(self, cells: List[int]):
 		# cells will already be padded up to numCells.
 		padded_cells = cells + [0] * (self._maxNumberOfCells - len(cells))
 		cellBytes = b"".join(intToByte(cell) for cell in padded_cells)
-		report = hwIo.hid.HidOutputReport(self._dev, reportID=self._cellValueCaps.ReportID)
-		report.setUsageValueArray(
-			HID_USAGE_PAGE_BRAILLE,
-			self._cellValueCaps.LinkCollection,
-			self._cellValueCaps.u1.NotRange.Usage,
-			cellBytes,
-		)
+		reportID = self._cellValueCaps[0].ReportID
+		report = hwIo.hid.HidOutputReport(self._dev, reportID=reportID)
+		for valueCap in self._cellValueCaps:
+			if valueCap.ReportID != reportID:
+				self._dev.write(report.data)
+				reportID = valueCap.ReportID
+				report = hwIo.hid.HidOutputReport(self._dev, reportID=reportID)
+			report.setUsageValueArray(
+				HID_USAGE_PAGE_BRAILLE,
+				valueCap.LinkCollection,
+				valueCap.u1.NotRange.Usage,
+				cellBytes[: valueCap.ReportCount],
```

Comment:
```suggestion
				cellBytes[:valueCap.ReportCount],
```

Reply:
I think these spaces were added by the formatter. pre-commit is failing without them.

Reply:
interesting, I would expect the opposite. Nevermind then